test/e2e/lazy_activation_policy_test.go

```diff
@@ -17,57 +17,241 @@ limitations under the License.
+			})
+		})
+
+		// Simple Case 1 (Policy created before resource)
+		// refer: https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/activation-preference/lazy-activation-preference.md#simple-case-1-policy-created-before-resource
+		ginkgo.It("Simple Case 1 (Policy created before resource)", func() {
+			ginkgo.By("step 1: deployment propagate success when policy created before it", func() {
+				waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+			})
+
+			ginkgo.By("step 2: after policy deleted, deployment still keep previous propagation states", func() {
+				framework.RemovePropagationPolicy(karmadaClient, namespace, policyName)
+				// wait to distinguish whether the state will not change or have no time to change
+				time.Sleep(pollInterval)
+				waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+			})
+		})
+
+		ginkgo.Context("Propagate dependencies", func() {
+			ginkgo.BeforeEach(func() {
+				policy.Spec.PropagateDeps = true
+				mountConfigMapToDeployment(deployment, configMapName)
+			})
+
+			ginkgo.JustBeforeEach(func() {
+				framework.CreateConfigMap(kubeClient, configMap)
+				ginkgo.DeferCleanup(func() {
+					framework.RemoveConfigMap(kubeClient, namespace, configMapName)
+				})
+			})
+
+			// Combined Case 5 (Propagate dependencies)
+			// refer: https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/activation-preference/lazy-activation-preference.md#combined-case-5-propagate-dependencies
+			ginkgo.It("Combined Case 5 (Propagate dependencies)", func() {
+				ginkgo.By("step 1: deployment and its dependencies could propagate success.", func() {
+					waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+					waitConfigMapPresentOnCluster(originalCluster, namespace, deploymentName)
+				})
+
+				ginkgo.By("step 2: change of lazy policy will not take effect", func() {
+					changePlacementTargetCluster(policy, modifiedCluster)
+					// wait to distinguish whether the policy will not take effect or have no time to take effect
+					time.Sleep(pollInterval)
+					waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+					waitConfigMapPresentOnCluster(originalCluster, namespace, deploymentName)
+				})
+
+				ginkgo.By("step 3: lazy policy take effect when deployment updated, dependencies can also been propagated.", func() {
+					updateDeploymentManually(deployment)
+					waitDeploymentPresentOnCluster(modifiedCluster, namespace, deploymentName)
+					waitConfigMapPresentOnCluster(modifiedCluster, namespace, deploymentName)
+				})
+			})
+		})
+	})
+
+	ginkgo.Context("2. Policy created after resource", func() {
+		ginkgo.JustBeforeEach(func() {
 			framework.CreateDeployment(kubeClient, deployment)
+			waitDeploymentExistInKarmada(namespace, deploymentName)
+			framework.CreatePropagationPolicy(karmadaClient, policy)
```

Comment:
do you have any better suggetion on how to ensure Policy reconciled after deployment? the above `waitDeploymentExistInKarmada` actually only  ensured Policy created after deployment. @XiShanYongYe-Chang  @zhzhuang-zju

Reply:
this is also may be the reason for ci failed: https://github.com/karmada-io/karmada/actions/runs/8585404537/job/23526833366?pr=4740

Reply:
Ignore the creation event and directly test the update event.

Reply:
Or just add a time delay? One second should be enough.

Reply:
updated.